### PR TITLE
Cleanup get_json_object params

### PR DESCRIPTION
**Pull request by revans2**
This cleans up how we pass parameters from get_json_object and multi-get_json_object to JNI.

It removes the callbacks into java from JNI to get the parameters and makes it consistent with other places in the code where we do similar things.

It also updates the types internally for `PathInstructionJni` to match the types used in the underlying C++ code. This lets us catch issues earlier and we can avoid passing more data than is needed.

I still plan on running a few performance tests just to be sure there is no impact.

### CODE REVIEW

**Comment by revans2** [comment]
build

File: src/main/cpp/src/JSONUtilsJni.cpp [hunk 1]
```diff
@@ -69,50 +66,50 @@ JNIEXPORT jlong JNICALL Java_com_nvidia_spark_rapids_jni_JSONUtils_getJsonObject
...
+    auto const type_nums = cudf::jni::native_jbyteArray(env, j_type_nums).to_vector();
+    auto const names     = cudf::jni::native_jstringArray(env, j_names);
+    auto const indexes   = cudf::jni::native_jintArray(env, j_indexes).to_vector();
     auto const num_paths = path_offsets.size() - 1;
     std::vector<path_type> paths(num_paths);
+    auto const num_entries = path_offsets[num_paths];
+
+    if (num_entries < 0 || names.size() != num_entries ||
+        indexes.size() != static_cast<uint64_t>(num_entries) ||
+        type_nums.size() != static_cast<uint64_t>(num_entries)) {
```
Comment (jlowe) [inline src/main/cpp/src/JSONUtilsJni.cpp:100]: Nit:
```suggestion
        indexes.size() != static_cast<std::size_t>(num_entries) ||
        type_nums.size() != static_cast<std::size_t>(num_entries)) {
```

**Review by jlowe — APPROVED** [review]

**Comment by revans2** [comment]
The results for my performance tests showed no meaningful difference, as expected.

**Review by jlowe — APPROVED** [review]

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
@jlowe please take another look

**Review by jlowe — APPROVED** [review]

**Pull request merged**